libraries/aws_ec2_security_group.rb

```diff
@@ -0,0 +1,87 @@
+class AwsEc2SecurityGroup < Inspec.resource(1)
+  name 'aws_ec2_security_group'
+  desc 'Verifies settings for an individual AWS Security Group.'
+  example '
+    describe aws_ec2_security_group("sg-12345678") do
+      it { should exist }
+    end
+  '
+
+  include AwsResourceMixin
+  attr_reader :description, :group_id, :group_name, :vpc_id
+
+  def to_s
+    'EC2 Security Group'
+  end
+
+  private
+
+  def validate_params(raw_params)
+    recognized_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:id, :group_id, :group_name, :vpc_id],
+      allowed_scalar_name: :group_id,
+      allowed_scalar_type: String,
+    )
+
+    # id is an alias for group_id
+    recognized_params[:group_id] = recognized_params.delete(:id) if recognized_params.key?(:id)
+
+    if recognized_params.key?(:group_id) && recognized_params[:group_id] !~ /^sg\-[0-9a-f]{8}/
+      raise ArgumentError, 'aws_ec2_security_group security group ID must be in the format "sg-" followed by 8 hexadecimal characters.'
+    end
+
+    if recognized_params.key?(:vpc_id) && recognized_params[:vpc_id] !~ /^vpc\-[0-9a-f]{8}/
+      raise ArgumentError, 'aws_ec2_security_group VPC ID must be in the format "vpc-" followed by 8 hexadecimal characters.'
+    end
+
+    validated_params = recognized_params
+
+    if validated_params.empty?
+      raise ArgumentError, 'You must provide parameters to aws_ec2_security_group, such as group_name, group_id, or vpc_id.g_group.'
+    end
+    validated_params
+  end
+
+  def fetch_from_aws
+    backend = AwsEc2SecurityGroup::BackendFactory.create
+
+    # Transform into filter format expected by AWS
+    filters = []
+    [
+      :description,
+      :group_id,
+      :group_name,
+      :vpc_id,
+    ].each do |criterion_name|
+      val = instance_variable_get("@#{criterion_name}".to_sym)
+      next if val.nil?
+      filters.push({
+                     name: criterion_name.to_s.tr('_', '-'),
+        values: [val],
```

Comment:
Should this be aligned with name? Just looks odd to me here.

Reply:
If you do anything reasonable, rubocop complains.  I may track down the offending rule and disable it; it's uglifying my hashes.